dvc/api.py

```diff
@@ -134,34 +177,16 @@ def _get_object_from_summon_file(name, path):
 
         return objects[0]
 
-    except FileMissingError:
-        raise SummonError("Summon file not found")
+    except FileNotFoundError as exc:
+        raise SummonError("Summon file not found") from exc
     except ruamel.yaml.YAMLError as exc:
         raise SummonError("Failed to parse summon file") from exc
     except Invalid as exc:
-        raise SummonError(str(exc))
-
-
-def _pull_dependencies(repo, deps):
-    if not deps:
-        return
-
-    outs = [repo.find_out_by_relpath(dep) for dep in deps]
-
-    with repo.state:
-        for out in outs:
-            repo.cloud.pull(out.get_used_cache())
-            out.checkout()
+        raise SummonError(str(exc)) from exc
 
 
+@wrap_with(threading.Lock())
 def _invoke_method(call, args, path):
-    # XXX: Some issues with this approach:
-    #   * Not thread safe
-    #   * Import will pollute sys.modules
```

Comment:
No message about `sys.path` manipulation?

Reply:
Reverted this considerations, but thread-safety.